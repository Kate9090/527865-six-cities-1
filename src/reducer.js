
const initialState = {
  city: `Amsterdam`,
  offerInCity: require(`./mocks/offers-in-amsterdam`).offerHotelList,
};


const ActionCreator = ({
  'NEW_CITY': (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),

  // 'RESET': () => ({
  //   type: `RESET`,
  // }),

});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return Object.assign({}, state, {
        city: action.payload,
        offerInCity: require(`./mocks/offers-in-${state.city}`).offer,
      });
    // case `RESET`:
    //   return Object.assign({}, state, initialState);
  }

  return state;
};

export {reducer, ActionCreator};
