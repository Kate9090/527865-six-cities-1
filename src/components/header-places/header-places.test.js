import React from "react";
import renderer from 'react-test-renderer';

import HeaderPlaces from './header-places.jsx';

import {Provider} from 'react-redux';
import reducer from '../../reducer/index';
import {createStore} from 'redux';

const store = createStore(
    reducer
);

import offerCities from '../../mocks/offers-city';
const mock = offerCities;

import offerHotelList from '../../mocks/offers-in-amsterdam';
const mockOffers = offerHotelList;

it(`renders correctly header places part`, () => {
  const tree = renderer
    .create(<Provider store={store}><HeaderPlaces
      activeCity = {mock[1]}
      offers= {mockOffers}
      cities={mock}
    /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
