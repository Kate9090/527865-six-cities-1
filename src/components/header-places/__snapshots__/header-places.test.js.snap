// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly header places part 1`] = `
Array [
  <h2
    className="visually-hidden"
  >
    Places
  </h2>,
  <b
    className="places__found"
  >
    0
     places to stay in 
  </b>,
]
`;
